tests/Unit/DataStructures/Tensor/Test_TensorData.py

```diff
@@ -0,0 +1,59 @@
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+from spectre.DataStructures import DataVector
+from spectre.DataStructures import ExtentsAndTensorVolumeData
+from spectre.DataStructures import TensorComponent
+import unittest
+import numpy as np
+import numpy.testing as npt
+
+
+class TestTensorData(unittest.TestCase):
+    # Tests for TensorComponent functions
+    def test_get_name(self):
+        tc = TensorComponent("tensor component", DataVector(1.0, 1.1))
+        self.assertEqual(tc.get_name(), "tensor component")
```

Comment:
You can combine all of these tests into two functions `test_tensor_component` and `test_extents_and_tensor_volume_data` I think, no need to split into so many functions.